lib/pluginLoader.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+module.exports = PluginLoader;
+
+function PluginLoader (Nocca) {
+
+    var self = this;
+
+    var loadedPlugins = {};
+    var instantiatedPlugins = {};
+
+    self.registerPlugins = registerPlugins;
+    self.instantiatePlugin = instantiatePlugin;
+
+    function registerPlugins (plugins) {
+
+        plugins.forEach(function (plugin) {
+
+            if (typeof plugin === 'string') {
+                // name for plugin!
+
+                // load from the plugins dir, should return an object
+                plugin = require(__dirname + '/plugins/' + plugin);
+
+            }
+
+
+            // if plugin was an object, or was require()'d as object, proceed
+            if (typeof plugin === 'object') {
+                // plugin definition!
+
+                registerPlugin(plugin);
+
+            }
+
+        });
+
+    }
+
+    function registerPlugin (pluginDefinition) {
+
+        // make sure it's valid before continuing
+        validatePluginDefinition(pluginDefinition);
+
+        // confirm plugin is not yet loaded
+        if (typeof loadedPlugins[pluginDefinition.id] === 'undefined') {
```

Comment:
As the plugins config property is an array at this time, it would be possible to redefine the same plugin twice. This could also be a result of config $extend resulting in overlap (if arrays are concatenated without uniqueness check). 

If double id's are ignored (as in id overlap would assume same plugin is referenced) this would be unnoticeable for happy flows, and unhappy flow would mean the custom-loaded plugin would not work as expected (if that one would be loaded later).

Could throw a warning maybe? Or hard-error and take a look at the config $extend function?